File: arduinobot_ws/src/arduinobot_py_examples/arduinobot_py_examples/simple_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SimplePublisher(Node):
    def __init__(self):
        #initalize the constructor of the Node with the name
        super().__init__("simple_publisher")

        #Create the publisher
        #We need to put the proper message type/Name of topic/buffer size in messages
        self.pub = self.create_publisher(String,"chatter",10)
        #Nbr of messages published in the topic
        self.counter = 0
        #Frequency of messages ex: 1 message/second
        self.frequency = 1.0
        self.get_logger().info("Publishing at %d Hz" % self.frequency)
        #it's a timer with the frequuency of 1hz where it calls the callback that we will define
        self.timer = self.create_timer(self.frequency,self.timerCallback)

    def timerCallback(self):
        #create a new message to be published
        msg = String()
        msg.data = "Hello Ros2 - counter: %d" % self.counter
        self.pub.publish(msg)
        self.counter += 1

def main():
    rclpy.init()
    simple_publisher = SimplePublisher()
    rclpy.spin(simple_publisher)
    simple_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: arduinobot_ws/src/arduinobot_cpp_examples/src/simple_publisher.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

#include <chrono>

using namespace std::chrono_literals;

//inhereting the Node class
class SimplePublisher : public rclcpp::Node{

public:
    SimplePublisher() : Node("simple_publisher"), counter_{0}
    {
        //Create the publisher with the name and number of message queue
        pub_ = create_publisher<std_msgs::msg::String>("chatterC",10);

        //binding the call back with the class instance,
        // and showing that the callback function is from the class
        timer_ = create_wall_timer(1s,std::bind(&SimplePublisher::timerCallBack,this));

        //Logger to output messages in kernel
        RCLCPP_INFO(get_logger(), "Publishing at 1 HZ");
    }

    void timerCallBack()
    {
        //creating a message variable to send
        auto message = std_msgs::msg::String();
        message.data = "Hello Ros2 - counter " + std::to_string(counter_++);
        pub_->publish(message);
    }

private:
    unsigned int counter_;
    //sharedptr is the pointer used to refer to the publisher and timer
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int args, char* argv[])
{
    rclcpp::init(args,argv);
    auto node = std::make_shared<SimplePublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0; 
}